pkg/cmd/openshift-controller-manager/cmd.go

```diff
@@ -79,21 +80,17 @@ func (o *OpenShiftControllerManager) Validate() error {
 }
 
 // StartControllerManager calls RunControllerManager and then waits forever
-func (o *OpenShiftControllerManager) StartControllerManager(stopCh <-chan struct{}) error {
-	if err := o.RunControllerManager(); err != nil {
+func (o *OpenShiftControllerManager) StartControllerManager(ctx context.Context) error {
+	if err := o.RunControllerManager(ctx); err != nil {
 		return err
 	}
 
 	go daemon.SdNotify(false, "READY=1")
-	select {
-	case <-stopCh:
-		klog.Infof("Controller Manager received stop signal. exiting.")
-		return nil
-	}
+	select {}
```

Comment:
this code reverts https://github.com/openshift/openshift-controller-manager/pull/215 a bit, but we should still terminate quickly on node reboot via `klog.Exit` by the lease's OnStoppedLeading